src/flows/shared/utils.ts

```diff
@@ -1,6 +1,34 @@
+import {Bucket} from 'src/types'
+
 export const formatQueryText = (queryText: string): string => {
   return (queryText || '')
     .trim()
     .split('|>')
     .join('\n  |>')
 }
+
+export const getNewestBucket = (
```

Comment:
this just signifies that the eventing isn't being passed around correctly right?